### PR TITLE
fix(gatsby): Fix an edge case with sift queries using index

### LINKED ISSUE #21457: Query running returns an incomplete result set in an edge case
## Description

Query running returns incomplete results in an edge case scenario when all of the following conditions are met:

1. A list of items is queried
2. The list is filtered by an `eq` filter (i.e. `{ field: { eq: "value" }`)
3. The field has a custom resolver (defined via schema customization)
4. The node doesn't have a value for this field

Originally reported in Twitter: https://twitter.com/jlengstorf/status/1228000066612293632

### Steps to reproduce

I will post a PR with a failing test soon.

### PR DESCRIPTION
**Pull request by vladar**
## Description

This PR fixes an edge case issue with indexed sift querying.

## Related Issues

Fixes #21457

### CODE REVIEW

**Comment by pvdz** [comment]
@vladar Thank you. I've pushed a commit that fixes this case, and hopefully the problem generically, by applying the fix we discussed; processing the filter "properly" earlier in the call stack, such that the fast indexes get built using those rather than the plain query args.

File: packages/gatsby/src/redux/run-sift.js [hunk 1]
```diff
@@ -41,10 +41,9 @@ const prepareQueryArgs = (filterFields = {}) =>
   }, {})
 
 const getFilters = filters =>
-  Object.keys(filters).reduce(
-    (acc, key) => acc.push({ [key]: filters[key] }) && acc,
-    []
-  )
+  Object.keys(filters).map(key => {
+    return { [key]: filters[key] }
+  })
```
Comment (vladar) [inline packages/gatsby/src/redux/run-sift.js]: Nice found!

**Comment by vladar** [comment]
It looks good to me, but the fun part is that I can't approve my own PR, so let's wait for @freiksenet approval %)

**Comment by pvdz** [comment]
Welll ..... 
image

File: packages/gatsby/src/redux/run-sift.js (hunk 1 above)
Comment (pvdz) [reply packages/gatsby/src/redux/run-sift.js]: Hm interesting. Is there a lint rule that changes `foo => ({x})` to `foo => { return {x} }` ? Cause that's what I had.

I'm not against this, just a little surprised :) I feel a little guilty for slipstreaming this one in but at the same time, shrug.

**Comment by pieh** [comment]
Published `gatsby@2.19.18`

**Pull request merged**